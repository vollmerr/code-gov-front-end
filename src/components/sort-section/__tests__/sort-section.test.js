import React from 'react';

import SortSection from 'components/sort-section'

import { shallow } from 'enzyme';


describe('Testing <SortSection /> component', () => {
  const props = {
    onSortChange: jest.fn(),
    options: [
      {label: 'test-a', value: 'a', selected: true},
      {label: 'test-b', value: 'b', selected: false}
    ]
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SortSection {...props} />);
  });
  it('renders SortSection', () => {
    expect(wrapper.exists()).toEqual(true);
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('Test changing sort', () => {
    wrapper.find('select').simulate('change', {target: {value: 'a'}});
    expect(wrapper.find("select").props().value).toBe("a");
  })
  it('Test onSortChange w/o props', () => {
    const props = {
      options: [
        {label: 'test-label', value: 'a', selected: true},
        {label: 'test-label2', value: 'b', selected: false}
      ]
    };
    wrapper = shallow(<SortSection {...props} />);
    wrapper.find('select').simulate('change', {target: {value: 'a'}});
    expect(wrapper.find("select").props().value).toBe("a");
  })
})