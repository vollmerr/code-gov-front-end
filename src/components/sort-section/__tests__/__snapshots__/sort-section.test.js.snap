// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SortSection /> component should render correctly 1`] = `
<div
  className="sort-section"
>
  <h2>
    <span>
      Sort by
    </span>
    <select
      aria-label="Sort Projects"
      onChange={[Function]}
      value="a"
    >
      <option
        aria-label="Sort Projects by test-a"
        key="a"
        value="a"
      >
        test-a
      </option>
      <option
        aria-label="Sort Projects by test-b"
        key="b"
        value="b"
      >
        test-b
      </option>
    </select>
  </h2>
</div>
`;
